docs/sql-distributed-sql-engine-spark-sql-cli.md

```diff
@@ -0,0 +1,165 @@
+---
+layout: global
+title: Spark SQL CLI
+displayTitle: Spark SQL CLI
+license: |
+  Licensed to the Apache Software Foundation (ASF) under one or more
+  contributor license agreements.  See the NOTICE file distributed with
+  this work for additional information regarding copyright ownership.
+  The ASF licenses this file to You under the Apache License, Version 2.0
+  (the "License"); you may not use this file except in compliance with
+  the License.  You may obtain a copy of the License at
+ 
+     http://www.apache.org/licenses/LICENSE-2.0
+ 
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+---
+
+* Table of contents
+{:toc}
+
+
+The Spark SQL CLI is a convenient tool to run the Hive metastore service in local mode and execute
+queries input from the command line. Note that the Spark SQL CLI cannot talk to the Thrift JDBC server.
+
+To start the Spark SQL CLI, run the following in the Spark directory:
+
+    ./bin/spark-sql
+
+Configuration of Hive is done by placing your `hive-site.xml`, `core-site.xml` and `hdfs-site.xml` files in `conf/`.
+
+## Spark SQL Command Line Options
+
+You may run `./bin/spark-sql --help` for a complete list of all available options.
+
+    CLI options:
+     -d,--define <key=value>          Variable substitution to apply to Hive
+                                      commands. e.g. -d A=B or --define A=B
+        --database <databasename>     Specify the database to use
+     -e <quoted-query-string>         SQL from command line
+     -f <filename>                    SQL from files
+     -H,--help                        Print help information
+        --hiveconf <property=value>   Use value for given property
+        --hivevar <key=value>         Variable substitution to apply to Hive
+                                      commands. e.g. --hivevar A=B
+     -i <filename>                    Initialization SQL file
+     -S,--silent                      Silent mode in interactive shell
+     -v,--verbose                     Verbose mode (echo executed SQL to the
+                                      console)
+
+## The hiverc File
+
+The Spark SQL CLI when invoked without the `-i` option will attempt to load `$HIVE_HOME/bin/.hiverc` and `$HOME/.hiverc` as initialization files.
+
+## Spark SQL CLI Interactive Shell Commands
+
+When `./bin/spark-sql` is run without either the `-e` or `-f` option, it enters interactive shell mode.
+Use `;` (semicolon) to terminate commands, but user can escape `;` by `\\;`. Comments in scripts can be specified using the `--` prefix.
```

Comment:
What if `;` is not the end of the line? does it still terminate commands?

Reply:
I think we should explain more about `;`
- what if it's inside a comment? `SELECT /* xxx; */ 1;`
- is `;` the only way to terminate commands? what if I type `SELECT 1` and then press enter?
- can we put multiple commands in a single line? `SELECT 1; SELECT 2;`

Reply:
How about current?